Analysis/Tasks/PWGCF/o2femtodream/femtoDreamProducerTask.cxx

```diff
@@ -0,0 +1,138 @@
+                                                      aod::Mults>>::iterator;
+using FilteredFullTracks = soa::Filtered<soa::Join<aod::FullTracks,
+                                                   aod::TracksExtended,
+                                                   aod::pidTPCEl, aod::pidTPCMu, aod::pidTPCPi,
+                                                   aod::pidTPCKa, aod::pidTPCPr, aod::pidTPCDe,
+                                                   aod::pidTOFEl, aod::pidTOFMu, aod::pidTOFPi,
+                                                   aod::pidTOFKa, aod::pidTOFPr, aod::pidTOFDe>>;
+} // namespace o2::aod
+
+struct femtoDreamProducerTask {
+
+  Produces<aod::FemtoDreamCollisions> outputCollision;
+  Produces<aod::FemtoDreamParticles> outputTracks;
+  Produces<aod::FemtoDreamDebugParticles> outputDebugTracks;
+
+  Configurable<bool> ConfDebugOutput{"ConfDebugOutput", true, "Debug output"};
+
+  /// Event cuts
+  FemtoDreamCollisionSelection colCuts;
+  Configurable<float> ConfEvtZvtx{"ConfEvtZvtx", 10.f, "Evt sel: Max. z-Vertex (cm)"};
+  Configurable<bool> ConfEvtTriggerCheck{"ConfEvtTriggerCheck", false, "Evt sel: check for trigger"};
+  Configurable<int> ConfEvtTriggerSel{"ConfEvtTriggerSel", kINT7, "Evt sel: trigger"};
+  Configurable<bool> ConfEvtOfflineCheck{"ConfEvtOfflineCheck", false, "Evt sel: check for offline selection"};
+
+  Filter colFilter = nabs(aod::collision::posZ) < ConfEvtZvtx;
+
+  FemtoDreamTrackSelection trackCuts;
+  Configurable<std::vector<float>> ConfTrkCharge{"ConfTrkCharge", std::vector<float>{-1, 1}, "Trk sel: Charge"};
+  Configurable<std::vector<float>> ConfTrkPtmin{"ConfTrkPtmin", std::vector<float>{0.4f, 0.6f, 0.5f, 0.2f}, "Trk sel: Min. pT (GeV/c)"};
+  Configurable<std::vector<float>> ConfTrkPtmax{"ConfTrkPtmax", std::vector<float>{4.05f, 999.f}, "Trk sel: Max. pT (GeV/c)"};
+  Configurable<std::vector<float>> ConfTrkEta{"ConfTrkEta", std::vector<float>{0.8f, 0.7f, 0.9f}, "Trk sel: Max. eta"};
+  Configurable<std::vector<float>> ConfTrkTPCnclsMin{"ConfTrkTPCnclsMin", std::vector<float>{80.f, 70.f, 90.f}, "Trk sel: Min. nCls TPC"};
+  Configurable<std::vector<float>> ConfTrkTPCfCls{"ConfTrkTPCfCls", std::vector<float>{0.7f, 0.83f, 0.9f}, "Trk sel: Min. ratio crossed rows/findable"};
+  Configurable<std::vector<float>> ConfTrkTPCsCls{"ConfTrkTPCsCls", std::vector<float>{0.1f, 160.f}, "Trk sel: Max. TPC shared cluster"};
+  Configurable<std::vector<float>> ConfTrkDCAxyMax{"ConfTrkDCAxyMax", std::vector<float>{0.1f, 999.f}, "Trk sel: Max. DCA_xy (cm)"}; /// here we need an open cut to do the DCA fits later on!
+  Configurable<std::vector<float>> ConfTrkDCAzMax{"ConfTrkDCAzMax", std::vector<float>{0.2f, 999.f}, "Trk sel: Max. DCA_z (cm)"};    /// open cut needed for v0s etc. also there we need a lower limit bit
+  /// \todo maybe we need to remove the PID from the general cut container and have a separate one, these are lots and lots of bits we need
```

Comment:
Would it be an idea that not all pid bits are supported in one sample? E.g. a maximum of three, but configurable?

Reply:
but what would be the benefit of this? 
In the end we don't want to have several data sets, and imagine you want to look at Sigma0 -> Lambda + Gamma, for which you need electrons (conversions), protons and pions (Lambda) and unfortunately the electrons are in your other data set. Then you add another data set?
Same goes for Xi's (p, pi, pi), Omegas (p, pi, K), ...

Reply:
How many species you think you put typically in each dataset?

Reply:
well, the five that we have here will be most relevant for us

Reply:
in any case, I'm considering to have an extra field for the PID identifiers (could be 0 for v0 etc. and this should compress well, right?) Or we store the children ID of v0, cascade etc. in there